File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        // HashSet<String> ls = new HashSet();
        // int count =0;
        // int left = 0;

        // for(int i=0;i<s.length();i++){
        //     while(int s[right]: ls){
        //         ls.remove(s[left]);
        //         left++;
        //     }
        //     ls.add(s[right]);
        //     count= math.max(count, right-left+1);
        // }
        // return count;  

        int maxlength = 0 ;
        for(int i=0;i<s.length();i++){
            StringBuilder currsub = new StringBuilder();
            for (int j = i; j < s.length();j++){
                if(currsub.indexOf(String.valueOf(s.charAt(j))) != -1 ){
                    break;
                }
                currsub.append(s.charAt(j));
                maxlength = Math.max(maxlength,currsub.length());

            }

        }
        return maxlength;

    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        # hp = set()
        # count = 0
        # l =0

        # for r in range(len(s)):
        #     while s[r] in hp:
        #         hp.remove(s[l])
        #         l+=1
        #     hp.add(s[r])
        #     count = max (count,r-l+1)
        # return count

        l=0
        longest=0
        dup =set()
        n = len(s)

        for r in range(n):
            while s[r] in dup:
                dup.remove(s[l])
                l+=1
            w =(r-l)+1
            longest = max(longest, w)
            dup.add(s[r])
        return longest
                